diff --git a/tinyxml2.cpp b/tinyxml2.cpp
index 69d93ef..cc013a5 100644
--- a/tinyxml2.cpp
+++ b/tinyxml2.cpp
@@ -86,7 +86,7 @@ distribution.
 	    if (!format) {
 	        return 0;
 	    }
-		int len = vsnprintf( 0, 0, format, va );
+		int len = vsnprintf( nullptr, 0, format, va );
 		TIXMLASSERT( len >= 0 );
 		return len;
 	}
@@ -168,8 +168,8 @@ void StrPair::TransferTo( StrPair* other )
     other->_end = _end;
 
     _flags = 0;
-    _start = 0;
-    _end = 0;
+    _start = nullptr;
+    _end = nullptr;
 }
 
 
@@ -179,8 +179,8 @@ void StrPair::Reset()
         delete [] _start;
     }
     _flags = 0;
-    _start = 0;
-    _end = 0;
+    _start = nullptr;
+    _end = nullptr;
 }
 
 
@@ -218,17 +218,17 @@ char* StrPair::ParseText( char* p, const char* endTag, int strFlags, int* curLin
         ++p;
         TIXMLASSERT( p );
     }
-    return 0;
+    return nullptr;
 }
 
 
 char* StrPair::ParseName( char* p )
 {
     if ( !p || !(*p) ) {
-        return 0;
+        return nullptr;
     }
     if ( !XMLUtil::IsNameStartChar( static_cast<unsigned char>(*p) ) ) {
-        return 0;
+        return nullptr;
     }
 
     char* const start = p;
@@ -247,7 +247,7 @@ void StrPair::CollapseWhitespace()
     // Adjusting _start would cause undefined behavior on delete[]
     TIXMLASSERT( ( _flags & NEEDS_DELETE ) == 0 );
     // Trim leading space.
-    _start = XMLUtil::SkipWhiteSpace( _start, 0 );
+    _start = XMLUtil::SkipWhiteSpace( _start, nullptr );
 
     if ( *_start ) {
         const char* p = _start;	// the read pointer
@@ -255,7 +255,7 @@ void StrPair::CollapseWhitespace()
 
         while( *p ) {
             if ( XMLUtil::IsWhiteSpace( *p )) {
-                p = XMLUtil::SkipWhiteSpace( p, 0 );
+                p = XMLUtil::SkipWhiteSpace( p, nullptr );
                 if ( *p == 0 ) {
                     break;    // don't write to q; this trims the trailing space.
                 }
